CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+
+    The first line is a real life imperative statement with a ticket number
+    from our issue tracker.  The body describes the behavior without the patch,
+    why this is a problem, and how the patch fixes the problem when applied.
+````
+
+* Make sure you have added the necessary tests for your changes.
+* Run _all_ the tests to assure nothing else was accidentally broken.
+
+## Making Trivial Changes
+
+### Documentation
+
+For changes of a trivial nature to comments and documentation, it is not
+always necessary to create a new ticket in Jira. In this case, it is
+appropriate to start the first line of a commit with '(doc)' instead of
+a ticket number.
+
+````
+    (doc) Add documentation commit example to CONTRIBUTING
+
+    There is no example for contributing a documentation commit
+    to the Puppet repository. This is a problem because the contributor
+    is left to assume how a commit of this nature may appear.
+
+    The first line is a real life imperative statement with '(doc)' in
+    place of what would have been the ticket number in a
+    non-documentation related commit. The body describes the nature of
+    the new documentation or comments added.
+````
+
+## Submitting Changes
+
+* Sign the [Contributor License Agreement](http://links.puppet.com/cla).
+* Push your changes to a topic branch in your fork of the repository.
+* Submit a pull request to the repository in the puppetlabs organization.
+* Update your Jira ticket to mark that you have submitted code and are ready for it to be reviewed (Status: Ready for Merge).
+  * Include a link to the pull request in the ticket.
+* The education team looks at Pull Requests on a regular basis.
+* After feedback has been given we expect responses within two weeks. After two
+  weeks we may close the pull request if it isn't showing any activity.
+
+## Revert Policy
+By running tests in advance and by engaging with peer review for prospective
+changes, your contributions have a high probability of becoming long lived
+parts of the the project. After being merged, the code will run through a
+series of testing pipelines on a large number of operating system
+environments. These pipelines can reveal incompatibilities that are difficult
+to detect in advance.
+
+If the code change results in a test failure, we will make our best effort to
+correct the error. If a fix cannot be determined and committed within 24 hours
+of its discovery, the commit(s) responsible _may_ be reverted, at the
+discretion of the committer and maintainers. This action would be taken to help 
+maintain passing states in our testing pipelines.
+
+The original contributor will be notified of the revert in the Jira ticket
+associated with the change. A reference to the test(s) and operating system(s)
+that failed as a result of the code change will also be added to the Jira
+ticket. This test(s) should be used to check future submissions of the code to
+ensure the issue has been resolved.
+
+## Release Policy
```

Comment:
Dumb question - the "release" branch was created for the Edu repositories. Is there any need for a separate "integration" branches where features are moved and testing is driven from before final push to the "release" branch? Maybe it's not necessary because releases aren't performed very often, but just wondered about it.